opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -413,18 +448,56 @@ def use_span(
     ) -> Iterator[trace_api.Span]:
         """See `opentelemetry.trace.Tracer.use_span`."""
         try:
-            span_snapshot = self._current_span_slot.get()
-            self._current_span_slot.set(span)
+            span_snapshot = self.source.get_current_span()
+            self.source._current_span_slot.set(  # pylint:disable=protected-access
+                span
+            )
             try:
                 yield span
             finally:
-                self._current_span_slot.set(span_snapshot)
+                self.source._current_span_slot.set(  # pylint:disable=protected-access
+                    span_snapshot
+                )
         finally:
             if end_on_exit:
                 span.end()
 
+
+class TracerSource(trace_api.TracerSource):
+    def __init__(
+        self,
+        sampler: sampling.Sampler = trace_api.sampling.ALWAYS_ON,
+        shutdown_on_exit: bool = True,
+    ):
+        # TODO: How should multiple TracerSources behave? Should they get their own contexts?
+        # This could be done by adding `str(id(self))` to the slot name.
```

Comment:
> Silencing noisy tracers from instrumented libraries is one of the main motivations for https://github.com/open-telemetry/oteps/blob/master/text/0016-named-tracers.md, and the span processor is only mechanism the API provides to suppress span creation.

I think I should actually add the instrumentation info to to the arguments for the sampler. What do you think?

Reply:
> Making TracerSource the shared container the tracer config (sampler, processor, etc.) and the context seems like it would create some problems.

But that's what we decided on in the spec, after some discussion: https://github.com/open-telemetry/opentelemetry-specification/issues/304 (and https://github.com/open-telemetry/opentelemetry-specification/issues/339)

Reply:
I would also argue that sharing the same trace configuration is generally the preferred thing to do.

Potentially there's value in adding additional processors per tracer, thus chaining the processing. But the vendor tracing integrations I've seen sample and process globally anyway.